File: servico-musica-java/java.soap/src/main/java/servico/java/soap/entity/Playlist.java
package servico.java.soap.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;

import java.util.ArrayList;
import java.util.List;

@Entity
public class Playlist {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private long id;
    private String nome;

    @ManyToMany
    private List<Musica> musicas = new ArrayList<>();

    @ManyToOne
    @JoinColumn(name = "usuario_id")
    @JsonIgnore
    private Usuario usuario;

    protected Playlist(){}

    public Playlist(String nome) {
        this.nome = nome;
    }

    public long getId() {
        return id;
    }

    public String getNome() {
        return nome;
    }

    public List<Musica> getMusicas() {
        return musicas;
    }

    public Usuario getUsuario() {
        return usuario;
    }

    public void setUsuario(Usuario usuario) {
        this.usuario = usuario;
    }
}

File: servico-musica-kotlin/kotlin.soap/src/main/kotlin/servico/kotlin/soap/entity/Playlist.kt
package servico.kotlin.soap.entity

import com.fasterxml.jackson.annotation.JsonIgnore
import jakarta.persistence.*

@Entity
class Playlist {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null

    var nome: String? = null

    @ManyToMany
    val musicas: MutableList<Musica?> = ArrayList()

    @ManyToOne
    @JoinColumn(name = "usuario_id")
    @JsonIgnore
    var usuario: Usuario? = null

    protected constructor()

    constructor(nome: String?) {
        this.nome = nome
    }
}